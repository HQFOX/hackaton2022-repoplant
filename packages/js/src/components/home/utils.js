import {
  Level1,
  Level2Average,
  Level3Bad,
  Level4,
  Level5
} from "@hv/uikit-react-icons";

export const getStatus = statusNumber => {
  switch (statusNumber) {
    case 1:
      return { Icon: Level1, sema: "sema10" };
    case 2:
      return { Icon: Level2Average, sema: "sema11" };
    case 3:
      return { Icon: Level3Bad, sema: "sema12" };
    case 4:
      return { Icon: Level4, sema: "sema13" };
    case 5:
      return { Icon: Level5, sema: "sema14" };
    default:
      return { Icon: null, sema: "sema1" };
  }
};
