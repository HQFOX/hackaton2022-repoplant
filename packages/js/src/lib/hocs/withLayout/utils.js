export const findById = (data = [], id) => {
  for (let i = 0; i < data.length; i += 1) {
    const el = data[i];
    if (el.id === id) return el;

    const child = findById(el.data, id);
    if (child) return child;
  }
  return null;
};

export const findLeaf = item => (item.data ? findLeaf(item.data[0]) : item);

export const findFirstChildByParentId = (data = [], id) => {
  const parent = findById(data, id);
  return (parent && parent.data && findLeaf(parent.data[0]).id) || id;
};

export const navigationDepth = arr =>
  Array.isArray(arr)
    ? 1 + Math.max(...arr.map(el => navigationDepth(el.data)))
    : 0;
