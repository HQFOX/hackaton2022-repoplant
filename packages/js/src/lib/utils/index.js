export const getRandom = (max, min = 1) =>
  Math.floor(Math.random() * (max - min + 1)) + min;
