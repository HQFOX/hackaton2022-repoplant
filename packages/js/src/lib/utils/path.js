const getSelection = (data, pathname) => {
  let selection;

  data.forEach(item => {
    if (item.path === pathname) selection = { ...item };
    if (item.data) getSelection(item.data, pathname);
  });

  return selection;
};

export { getSelection };
