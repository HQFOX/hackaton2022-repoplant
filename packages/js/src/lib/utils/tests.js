import React from "react";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { createMemoryHistory } from "history";
import HvProvider from "@hv/uikit-react-core/dist/Provider";

const mockStore = configureMockStore([thunk]);
const memoryHistory = createMemoryHistory();

const createMockStore = mockState => mockStore(mockState);

const withStoreProvider = (Component, store) => props => (
  <Provider store={createMockStore(store)}>
    <HvProvider>
      <MemoryRouter>
        <Component {...props} />
      </MemoryRouter>
    </HvProvider>
  </Provider>
);

const withHvProvider = Component => props => (
  <HvProvider>
    <MemoryRouter>
      <Component {...props} />
    </MemoryRouter>
  </HvProvider>
);

const withProps = Component => props => <Component {...props} />;

export {
  createMockStore,
  mockStore,
  memoryHistory,
  withStoreProvider,
  withHvProvider,
  withProps
};
