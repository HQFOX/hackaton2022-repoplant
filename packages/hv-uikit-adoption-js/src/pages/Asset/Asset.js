import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import HvTypography from "@hv/uikit-react-core/dist/Typography";
import HvGrid from "@hv/uikit-react-core/dist/Grid";
import withLayout from "lib/hocs/withLayout";
import { fetchAsset } from "lib/api/assets";

const Asset = () => {
  const [data, setData] = useState({});
  const { id } = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const result = await fetchAsset(id);
      setData(result);
    };
    fetchData();
  });

  return (
    <HvGrid container>
      <HvGrid item xl={12}>
        <HvTypography variant="3xlTitle">{data.headerTitle}</HvTypography>
        <br />
        <HvTypography variant="normalText">{data.relatedAssets}</HvTypography>
      </HvGrid>
    </HvGrid>
  );
}

export default withLayout(Asset);
