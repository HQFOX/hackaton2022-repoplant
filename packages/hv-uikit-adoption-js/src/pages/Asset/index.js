import withStyles from "@material-ui/core/styles/withStyles";
import styles from "./styles";
import Asset from "./Asset";

export default withStyles(styles, { name: "Asset", withTheme: true })(
  Asset
);
