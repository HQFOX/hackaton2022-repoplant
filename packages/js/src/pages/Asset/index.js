import { withStyles } from "@material-ui/core";
import styles from "./styles";
import Asset from "./Asset";

export default withStyles(styles, { name: "Asset" })(Asset);
