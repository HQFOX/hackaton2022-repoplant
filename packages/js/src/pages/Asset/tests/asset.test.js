import React from "react";
import { mount } from "enzyme";
import { withStoreProvider } from "lib/utils/tests";
import Asset from "..";

describe("<Asset />", () => {
  let component;

  const WithStoreProvider = withStoreProvider(Asset, {
    router: { location: { pathname: "/" } },
    auth: { isAuthed: false },
    pages: { data: [] },
    logout: jest.fn()
  });

  beforeEach(() => {
    component = mount(<WithStoreProvider />);
  });

  it("it matches the snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
