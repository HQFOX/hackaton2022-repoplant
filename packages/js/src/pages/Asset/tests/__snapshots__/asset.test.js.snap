// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Asset /> it matches the snapshot 1`] = `undefined`;
